chain/chunks/src/lib.rs

```diff
@@ -607,11 +607,46 @@ impl ShardsManager {
             .collect::<HashSet<_>>()
     }
 
+    /// Check if chunk forwarding is enabled
+    /// It is enabled if
+    /// 1) the node is a validator in the current epoch
+    /// 2) feature ForwardChunkParts is enabled
+    fn is_chunk_forwarding_enabled(&self, prev_hash: &CryptoHash) -> Result<bool, Error> {
+        // chunks will not be forwarded to non-validators
+        if self.me.is_none() {
+            return Ok(false);
+        }
+        let epoch_id = self.runtime_adapter.get_epoch_id_from_prev_block(prev_hash)?;
+        let protocol_version = self.runtime_adapter.get_epoch_protocol_version(&epoch_id)?;
+        if !checked_feature!("stable", ForwardChunkParts, protocol_version) {
+            return Ok(false);
+        }
+        Ok(self
+            .runtime_adapter
+            .get_validator_by_account_id(&epoch_id, prev_hash, self.me.as_ref().unwrap())
+            .is_ok())
+    }
+
+    /// Only marks this chunk as being requested
+    /// Note no requests are actually sent at this point.
+    fn request_chunk_single_mark_only(&mut self, chunk_header: &ShardChunkHeader) {
+        self.request_chunk_single(chunk_header, None, false)
+    }
+
+    /// send partial chunk requests for one chunk
+    /// `chunk_header`: the chunk being requested
+    /// `ancestor_hash`: hash of an ancestor block of the block the chunk belongs to. It must satisfy
```

Comment:
but it is not an argument to this function...

Reply:
whoops this was from the other PR I was working on. Must have been carried over when I was splitting them